cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -216,6 +218,63 @@ auto batched_compress_temp_size(compression_type compression,
   return temp_size;
 }
 
+#if NVCOMP_HAS_COMP_TEMPSIZE_EX(NVCOMP_MAJOR_VERSION, NVCOMP_MINOR_VERSION, NVCOMP_PATCH_VERSION)
+// Wrapper for nvcompBatched<format>CompressGetTempSizeEx
+auto batched_compress_get_temp_size_ex(compression_type compression,
+                                       size_t batch_size,
+                                       size_t max_uncompressed_chunk_bytes,
+                                       size_t max_total_uncompressed_bytes)
+{
+  size_t temp_size             = 0;
+  nvcompStatus_t nvcomp_status = nvcompStatus_t::nvcompSuccess;
+  switch (compression) {
+    case compression_type::SNAPPY:
+      nvcomp_status = nvcompBatchedSnappyCompressGetTempSizeEx(batch_size,
+                                                               max_uncompressed_chunk_bytes,
+                                                               nvcompBatchedSnappyDefaultOpts,
+                                                               &temp_size,
+                                                               max_total_uncompressed_bytes);
+      break;
+    case compression_type::DEFLATE:
+      nvcomp_status = nvcompBatchedDeflateCompressGetTempSizeEx(batch_size,
+                                                                max_uncompressed_chunk_bytes,
+                                                                nvcompBatchedDeflateDefaultOpts,
+                                                                &temp_size,
+                                                                max_total_uncompressed_bytes);
+      break;
+    case compression_type::ZSTD:
+      nvcomp_status = nvcompBatchedZstdCompressGetTempSizeEx(batch_size,
+                                                             max_uncompressed_chunk_bytes,
+                                                             nvcompBatchedZstdDefaultOpts,
+                                                             &temp_size,
+                                                             max_total_uncompressed_bytes);
+      break;
+    default: CUDF_FAIL("Unsupported compression type");
+  }
+
+  CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess,
+               "Unable to get scratch size for compression");
+  return temp_size;
+}
+#endif
+
+size_t batched_compress_temp_size(compression_type compression,
+                                  size_t num_chunks,
+                                  size_t max_uncomp_chunk_size,
+                                  size_t max_total_uncomp_size)
+{
+#if NVCOMP_HAS_COMP_TEMPSIZE_EX(NVCOMP_MAJOR_VERSION, NVCOMP_MINOR_VERSION, NVCOMP_PATCH_VERSION)
+  // Ignore errors in the expanded version; fall back to the old API in case of failure
+  try {
+    return batched_compress_get_temp_size_ex(
+      compression, num_chunks, max_uncomp_chunk_size, max_total_uncomp_size);
+  } catch (...) {
+  }
```

Comment:
You catch but you don't throw so the code will continue even there was exception thrown. Is this expected? If so please add a comment to clarify that.

Reply:
There's a comment above the try block. Should it be placed elsewhere?

Reply:
Oh sorry I really didn't see it. So I think the comment should be inside the `catch` block instead.